Biology.java

```diff
@@ -0,0 +1,346 @@
+import cern.colt.Arrays;
+
+import java.io.File;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Scanner;
+
+/**
+ * A class that allows antigen to model mutations more realistically, thus, respecting biology
+ *
+ * @author Thien Tran
+ */
+public class Biology {
+    public enum DMSData {
+        DMS_DATA();
+        public final double[][] aminoAcidPreference;
+
+        DMSData() {
+            int numberOfAminoAcidSites = Parameters.startingSequence.length() / 3;
+            int numberOfAminoAcids = Parameters.AlphabetType.AMINO_ACIDS.getValidCharacters().length();
+            aminoAcidPreference = new double[numberOfAminoAcidSites][numberOfAminoAcids];
+
+            try {
+                Scanner dms = new Scanner(new File(Parameters.DMSData));
+                dms.nextLine(); // read header
+
+                for (int i = 0; i < numberOfAminoAcidSites; i++) {
+                    Scanner currentSite = new Scanner(dms.nextLine());
+                    currentSite.useDelimiter(",");
+
+                    currentSite.next(); // ignore site number
+
+                    for (int j = 0; j < numberOfAminoAcids; j++) {
+                        double probability = currentSite.nextDouble();
+                        aminoAcidPreference[i][j] = probability;
+                    }
+                }
+            } catch (Exception e) {
+                e.printStackTrace();
+            }
+        }
+
+        public double[][] getAminoAcidPreference() {
+            return aminoAcidPreference;
+        }
+    }
+
+    /**
+     * Possible mutations that can occur at a single nucleotide site
```

Comment:
Could we be more descriptive here? Like: "Code for drawing nucleotide mutations from a probability distribution, where each possible mutation is weighted by a pre-defined transition/transversion ratio."